client/state/analytics/middleware.js

```diff
@@ -32,45 +32,45 @@ const eventServices = {
 	adwords: ( { properties } ) => trackCustomAdWordsRemarketingEvent( properties ),
 };
 
-// Whitelists specific parameters to avoid polluting page view events
-const PAGE_VIEW_SERVICES_ALLOWED_PARAMS = [ 'client_id' ];
-
 const pageViewServices = {
 	ga: ( { url, title } ) => analytics.ga.recordPageView( url, title ),
-	default: ( { url, title, ...params } ) =>
-		analytics.pageView.record( url, title, pick( params, PAGE_VIEW_SERVICES_ALLOWED_PARAMS ) ),
+	default: ( { url, title, ...params } ) => analytics.pageView.record( url, title, params ),
 };
 
-const loadTrackingTool = ( trackingTool, state ) => {
+const loadTrackingTool = ( { trackingTool }, state ) => {
 	if ( trackingTool === 'HotJar' && ! isTracking( state, 'HotJar' ) ) {
 		analytics.hotjar.addHotJarScript();
 	}
 };
 
 const statBump = ( { group, name } ) => analytics.mc.bumpStat( group, name );
 
+const setAnonymousUserId = ( { anonId } ) => analytics.tracks.setAnonymousUserId( anonId );
+
+const setOptOut = ( { isOptingOut } ) => analytics.tracks.setOptOut( isOptingOut );
```

Comment:
We also won't need these helper methods unless we want to normalize the payload before going through the switch.

Reply:
I think those are used to transform the "object-payload" to a raw type since we cannot use raw types as `payload` anymore. I'm not entirely sure whether adding another layer is a good idea though: it might be worth updating the entire chain of calls to handle directly `{ isOptingOut }`. ( I'm not 100% sure but I think `setAnonymousUserId` is basically dead code at this point )